modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -535,8 +534,11 @@ private CompletableFuture<Void> createZoneReplicationNodes(
             int catalogVersion,
             int partitionCount
     ) {
-        return inBusyLockAsync(busyLock, () -> {
-            assert assignments != null : IgniteStringFormatter.format("Zone has empty assignments [id={}].", zoneId);
+        // Zone nodes might be created in the same catalog version as a table in this zone, so there could be a race
+        // between storage creation due to replica start and storage creation due to table addition. To eliminate the race,
+        // we acquire a write lock on the zone (table addition acquires a read lock).
+        return inBusyLockAsync(busyLock, () -> executeUnderZoneWriteLock(zoneId, () -> inBusyLockAsync(busyLock, () -> {
+        assert assignments != null : IgniteStringFormatter.format("Zone has empty assignments [id={}].", zoneId);
```

Comment:
Something is not right with this formatting

Reply:
```
inBusyLockAsync(busyLock, () -> executeUnderZoneWriteLock(zoneId, () -> inBusyLockAsync(busyLock, 
```

This looks so awkward...

Reply:
Tried to fix it